### PR TITLE
Refactor StringWriters to use our new MessageBuilder

### PR DESCRIPTION
**Pull request by nielsm5**
Refactor all code which uses PathMessage.asTemporaryMessage
Convert some XmlWriter instances

### CODE REVIEW

File: core/src/main/java/org/frankframework/processors/InputOutputPipeProcessor.java [hunk 1]
```diff
@@ -166,8 +166,16 @@ private void processMessageCompaction(IPipe pipe, PipeLineSession pipeLineSessio
...
-		CompactSaxHandler handler = new CompactSaxHandler(xmlWriter);
+
+		final MessageBuilder messageBuilder;
+		try {
+			messageBuilder = new MessageBuilder();
+		} catch (IOException e) {
+			log.warn("Pipeline of adapter [{}] could not compact received message", owner.getName(), e);
+			return;
+		}
+
```
Comment (tnleeuw) [inline core/src/main/java/org/frankframework/processors/InputOutputPipeProcessor.java:177]: Why not put everything in the try-catch clause below?
(I don't like the way it looks with this separate try-catch here).

**Review by tnleeuw — APPROVED** [review]

**Comment by nielsm5** [comment]
Accidently pushed a commit to the wrong branch, but as the changeset was relatively small I've merged it entirely...

File: core/src/main/java/org/frankframework/stream/document/DocumentBuilderFactory.java [hunk 2]
```diff
@@ -38,9 +39,9 @@ public static IDocumentBuilder startDocument(DocumentFormat format, String rootE
 	public static IDocumentBuilder startDocument(DocumentFormat format, String rootElement) throws SAXException {
 		switch (format) {
 		case XML:
-			return new XmlDocumentBuilder(rootElement);
+			return new XmlDocumentBuilder(rootElement, new StringWriter());
 		case JSON:
-			return new JsonDocumentBuilder();
+			return new JsonDocumentBuilder(new StringWriter());
```
Comment (nielsm5) [inline core/src/main/java/org/frankframework/stream/document/DocumentBuilderFactory.java:44]: We need to migrate away from this at some point, for now this is the only place this 'evil' in memory writer is used.

File: filesystem/src/main/java/org/frankframework/filesystem/FileSystemActor.java [hunk 3]
```diff
@@ -503,27 +506,29 @@ private interface FileAction<F> {
...
-			ArrayBuilder directoryBuilder = DocumentBuilderFactory.startArrayDocument(getOutputFormat(), action+"FilesList", "file");
-			try(Stream<F> stream = FileSystemUtils.getFilteredStream(fileSystem, folder, getWildcard(), getExcludeWildcard(), TypeFilter.FILES_ONLY)) {
-				Iterator<F> it = stream.iterator();
-				while(it.hasNext()) {
-					F file = it.next();
-					try (INodeBuilder nodeBuilder = directoryBuilder.addElement()){
-						FileSystemUtils.getFileInfo(fileSystem, file, nodeBuilder);
-						action.execute(file);
+			MessageBuilder messageBuilder = new MessageBuilder();
+			try (ArrayBuilder directoryBuilder = DocumentBuilderFactory.startArrayDocument(getOutputFormat(), action+"FilesList", "file", messageBuilder, true)) {
```
Comment (nielsm5) [inline filesystem/src/main/java/org/frankframework/filesystem/FileSystemActor.java:513]: This was never closed?!

File: core/src/main/java/org/frankframework/stream/OverflowToDiskOutputStream.java [hunk 4]
```diff
@@ -90,29 +106,76 @@ public void write(byte[] b, int off, int len) throws IOException {
...
 			outputStream.write(b, off, len);
 			return;
 		}
 
 		// Write to the buffer
-		System.arraycopy(b, off, buffer, count, len);
-		count += len;
+		currentBufferSize += len;
+		while (len > 0) {
+			BufferBlock s = last();
```
Comment (tnleeuw) [inline core/src/main/java/org/frankframework/stream/OverflowToDiskOutputStream.java:120]: Ik zou een reference naar het laatste BufferBlock vasthouden, als optimalisatie, want ik vermoed dat je die heel vaak nodig hebt.

**Review by tnleeuw — APPROVED** [review]

**Pull request merged**